File: Programming/Java/MedianOfTwoSortedArrays/MedianOfTwoSortedArrays.java
import java.util.Scanner;
 
public class MedianOfTwoSortedArrays
 
{   public static void main(String[] args) 
    {
        Scanner scan = new Scanner(System.in);
        System.out.println("Enter number of elements in arrays");
        int N = scan.nextInt();
        int[] nums1 = new int[ N ];
        int[] nums2 = new int[ N ];
        System.out.println("Enter "+ N +" elements of array 1");
        for (int i = 0; i < N; i++)
            nums1[i] = scan.nextInt();
        System.out.println("Enter "+ N +" elements of array 2");
        for (int i = 0; i < N; i++)
            nums2[i] = scan.nextInt();
        int med = median(nums1, nums2);
        System.out.println("Median = "+ med);
     }
     public static int median(int[] nums1, int[] nums2)
     {
         int N = nums1.length;
         return median(nums1, 0, N -1 , nums2, 0, N - 1);
     }
     public static int median(int[] nums1, int l1, int h1, int[] nums2, int l2, int h2)
     {
         int mid1 = (h1 + l1 ) / 2;
         int mid2 = (h2 + l2 ) / 2;
 
         if (h1 - l1 == 1)
             return (Math.max(nums1[l1] , nums2[l2]) + Math.min(nums1[h1] , nums2[h2]))/2;
         else if (nums1[mid1] > nums2[mid2])
             return median(nums1, l1, mid1 , nums2, mid2 , h2);    
         else
             return median(nums1, mid1 , h1, nums2, l2 , mid2 );    
     }     
}

File: Programming/Python/MedianOfTwoSortedArrays/median_of_two_sorted_arrays.py


          
def find_median_of_merged_version_of_two_sorted_array(nums1,nums2):
    i =0
    j=0
    k=0
    merged = [ 0 for i in range(len(nums1)+len(nums2))]
    while i < len(nums1) and j < len(nums2): 
        if nums1[i] < nums2[j]: 
            merged[k] = nums1[i] 
            i+= 1
        else: 
            merged[k] = nums2[j] 
            j+= 1
        k+= 1
        
    while i < len(nums1): 
        merged[k] = nums1[i] 
        i+= 1
        k+= 1
        
    while j < len(nums2): 
        merged[k] = nums2[j] 
        j+= 1
        k+= 1
          
    
    if len(merged)%2==0:
        _1=len(merged)/2-1
        _2=len(merged)/2
        return (merged[ _1]+merged[_2] )/2
    else:
        _1=int((len(merged)-1)/2)
        return merged[_1 ]
          
        
find_median_of_merged_version_of_two_sorted_array([1,3],[2])